go/registry/api/api.go

```diff
@@ -930,6 +937,43 @@ func VerifyRegisterRuntimeArgs(logger *logging.Logger, sigRt *SignedRuntime, isG
 	return &rt, nil
 }
 
+// VerifyRegisterComputeRuntimeArgs verifies ComputeRuntime-specific arguments for RegisterRuntime.
+func VerifyRegisterComputeRuntimeArgs(logger *logging.Logger, rt *Runtime, runtimeLookup RuntimeLookup) error {
+	// Check runtime's key manager, if key manager ID is set.
+	if rt.KeyManager != nil {
+		km, err := runtimeLookup.Runtime(*rt.KeyManager)
+		if err != nil {
+			logger.Error("RegisterRuntime: error when fetching the runtime's key manager from registry",
+				"runtime", rt.ID,
+				"key_manager", rt.KeyManager,
+			)
+			return err
+		}
+
+		// Key manager runtime must exist.
+		if km == nil {
```

Comment:
I think that `Runtime()` should return `ErrNoSuchRuntime` if the runtime doesn't exist, so this additional check is not needed?